myokit/_sim/pacing.h

```diff
@@ -750,21 +754,31 @@ FSys_Destroy(FSys sys)
  * Returns a fixed-form pacing error flag.
  */
 FSys_Flag
-FSys_Populate(FSys sys, PyObject* times_list, PyObject* values_list)
+FSys_Populate(FSys sys, PyObject* protocol)
 {
     int i;
     Py_ssize_t n;
 
     // Check ESys
     if(sys == 0) return FSys_INVALID_SYSTEM;
     if (sys->n_points != -1) return FSys_POPULATED_SYSTEM;
-
-    // Check input lists
-    if(!PyList_Check(times_list)) return FSys_POPULATE_INVALID_TIMES;
-    if(!PyList_Check(values_list)) return FSys_POPULATE_INVALID_VALUES;
+    if (protocol == Py_None) return FSys_POPULATE_INVALID_PROTOCOL;
+
+    // Get PyList from protocol (will need to decref!)
+    // Cast to (char*) happens because CallMethod accepts a mutable char*
+    // This should have been const char* and has been fixed in python 3
+    PyObject* times_list = PyObject_CallMethod(protocol, (char*)"times", NULL); // Returns a new reference
+    if(times_list == NULL) return FSys_POPULATE_INVALID_PROTOCOL;
+    if(!PyList_Check(times_list)) {
+        Py_DECREF(times_list);
```

Comment:
Do `times_list` and `values_list` need to be decreffed on every possible exit of this function after this? Looks like a dangling reference otherwise?

Reply:
Fixed it!